### PR TITLE
warehouse fixes

### PR DESCRIPTION
**Pull request by jordienr**

### CODE REVIEW

File: apps/studio/components/interfaces/DataWarehouse/WarehouseMenuItem.tsx [hunk 1]
```diff
@@ -99,6 +99,9 @@ export const WarehouseMenuItem = ({ item }: Props) => {
         <DropdownMenu>
           <DropdownMenuTrigger asChild>
             <Button
+              onClick={(e) => {
+                e.preventDefault()
```
Comment (jordienr) [inline apps/studio/components/interfaces/DataWarehouse/WarehouseMenuItem.tsx:103]: prevents navigating to the collection if you're opening the dropdown

File: apps/studio/components/interfaces/DataWarehouse/WarehouseCollectionDetail.tsx [hunk 2]
```diff
@@ -68,15 +67,15 @@ export const WarehouseCollectionDetail = () => {
     offset: 0,
     search: '',
     interval: {
-      to: INTERVALS[0].calcTo(),
-      from: INTERVALS[0].calcFrom(),
+      to: () => INTERVALS[0].calcTo(),
+      from: () => INTERVALS[0].calcFrom(),
```
Comment (charislam) [inline apps/studio/components/interfaces/DataWarehouse/WarehouseCollectionDetail.tsx:71]:
```suggestion
      to: INTERVALS[0].calcTo,
      from: INTERVALS[0].calcFrom,
```

tiniest nit

File: apps/studio/components/interfaces/DataWarehouse/WarehouseCollectionDetail.tsx [hunk 3]
```diff
@@ -128,10 +125,16 @@ order by timestamp desc limit ${filters.limit} offset ${filters.offset}
...
-    e.preventDefault()
-    setFilters({ ...filters, search })
-    refetch()
+  function refreshResults(e?: React.FormEvent<HTMLFormElement>) {
+    e?.preventDefault()
+    setFilters({
+      ...filters,
+      search,
+      interval: {
+        to: () => dayjs().toISOString(),
```
Comment (charislam) [inline apps/studio/components/interfaces/DataWarehouse/WarehouseCollectionDetail.tsx:134]: Looking at this, looks like the intended effect and maybe the bug you're fixing in the first place 😄 but I found myself a little confused when I had selected a concrete date in the past, and refreshing changed the selected time period.

I realize that's confusing so here's what I mean.

What I expect to happen:

(1) If I have an open-ended interval to current time (e.g., last hour, last week, date range selected that includes the current date), I expect refreshing to update the end time for that range to the current time, which it does 👍🏼 

(2) But if I have a close-ended interval ending at some time in the past (e.g., September 3 - September 4, today being September 11), I don't expect refreshing to change my interval to September 3 - September 11

Not a blocker and happy with it if this is the intended UX 😄

**Review by charislam — APPROVED** [review]
Minor UI feedback but no blockers. Docs build error is unrelated, I will fix 😄 

Tested by creating and refreshing warehouse collection

**Pull request merged**